clients/src/main/java/org/apache/kafka/common/metrics/MetricConfig.java

```diff
@@ -65,6 +69,15 @@ public MetricConfig timeWindow(long window, TimeUnit unit) {
         return this;
     }
 
+    public Map<String, String> tags() {
```

Comment:
Currently, we add the default tag (e.g, client-id) explicitly for each metric. With the default tags, I am wondering if it's simpler to add the following method in Metrics that automatically adds the default tags in the metric name. Then, we can use that method to create the metric name and remove the code that explicitly sets the default tags.

MetricName metricName(...)